unittests/API/APITest.cpp

```diff
@@ -433,6 +433,24 @@ TEST(HermesWatchTimeLimitTest, WatchTimeLimit) {
   }
 }
 
+TEST(HermesTriggerAsyncTimeoutTest, TriggerAsyncTimeout) {
+  // Some code that exercies the async break checks.
+  const char *forEver = "for (;;){}";
+  uint32_t ShortTimeoutMS = 123;
+  {
+    // Single runtime with ~20 minute limit that will not be reached.
+    auto rt = makeHermesRuntime();
+    std::thread t([&]() {
+      std::this_thread::sleep_for(std::chrono::milliseconds(ShortTimeoutMS));
+      rt->asyncTriggerTimeout();
+    });
+    t.detach();
```

Comment:
Why detach the thread here?

Reply:
I don’t know. What’s the alternative?

Reply:
The tests failed without it. My understanding is `.join` won’t work because it’ll wait for the thread to finish - so we’ll end up in deadlock. But I didn’t actually try it so I could be wrong

Reply:
We should be able to `join` after the `ASSERT_THROW`.

Reply:
You're right - that works. What happened here? Without `join` and `detach`, it did not work. But to my understanding, the `ASSERT_THROW` won't return unless the thread executes the code to stop the runtime. But by that time, the `.join` shouldn't do anything, since we basically know the thread code has run. How does the `.join` afterwards make any impact here?

Reply:
From the reference for `joinable` ([link](https://en.cppreference.com/w/cpp/thread/thread/joinable)):

> A thread that has finished executing code, but has not yet been joined is still considered an active thread of execution and is therefore joinable.

And for `~thread` ([link](https://en.cppreference.com/w/cpp/thread/thread/~thread)):

> If `*this` has an associated thread (`joinable() == true`), `std::terminate()` is called.

So even if the thread has finished executing, it should be joined before it is destroyed. That said, that there is actually no guarantee that the thread has actually finished executing. All we know when the exception is thrown is that the interrupt flag in the runtime was set. In theory, the thread could have set the flag, and then immediately been suspended before it could return.